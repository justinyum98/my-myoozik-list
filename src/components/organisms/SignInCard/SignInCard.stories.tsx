import { ComponentStory, ComponentMeta } from '@storybook/react';

import SignInCard from './SignInCard';

export default {
  title: 'SignInCard',
  component: SignInCard,
} as ComponentMeta<typeof SignInCard>;

export const Primary: ComponentStory<typeof SignInCard> = () => (
  <SignInCard onSignIn={() => console.log('signed in')} />
);
