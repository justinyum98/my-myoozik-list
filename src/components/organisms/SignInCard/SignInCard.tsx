import Paper from '@mui/material/Paper';
import Typography from '@mui/material/Typography';
import Button from '@mui/material/Button';

type SignInCardProps = {
  onSignIn: () => void;
};

const SignInCard = ({ onSignIn }: SignInCardProps) => (
  <Paper
    variant="outlined"
    sx={{
      display: 'flex',
      flexDirection: 'column',
      justifyContent: 'center',
      alignItems: 'center',
      maxWidth: 600,
      paddingX: 9,
      paddingY: 8,
    }}
  >
    <Typography variant="h4" gutterBottom>
      My Myoozik List
    </Typography>
    <Button onClick={onSignIn} variant="contained" fullWidth sx={{ mt: 4 }}>
      Sign In
    </Button>
  </Paper>
);

export default SignInCard;
