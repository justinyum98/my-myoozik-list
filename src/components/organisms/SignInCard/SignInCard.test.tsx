import { render, fireEvent, screen } from '@testing-library/react';

import SignInCard from './SignInCard';

const mockOnSignIn = jest.fn();

describe('<SignInCard /> organism component', () => {
  beforeEach(() => {
    mockOnSignIn.mockClear();
  });

  it('should render', () => {
    const { asFragment } = render(<SignInCard onSignIn={mockOnSignIn} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('should call onSignIn event handler on button click', () => {
    render(<SignInCard onSignIn={mockOnSignIn} />);

    fireEvent.click(screen.getByText('Sign In'));

    expect(mockOnSignIn).toBeCalledTimes(1);
  });
});
