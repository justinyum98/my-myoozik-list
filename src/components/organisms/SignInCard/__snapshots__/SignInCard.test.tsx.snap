// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SignInCard /> organism component should render 1`] = `
<DocumentFragment>
  <div
    class="MuiPaper-root MuiPaper-outlined MuiPaper-rounded css-z0fmtq-MuiPaper-root"
  >
    <h4
      class="MuiTypography-root MuiTypography-h4 MuiTypography-gutterBottom css-1vw6mcs-MuiTypography-root"
    >
      My Myoozik List
    </h4>
    <button
      class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-fullWidth MuiButtonBase-root  css-1hny3at-MuiButtonBase-root-MuiButton-root"
      tabindex="0"
      type="button"
    >
      Sign In
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </div>
</DocumentFragment>
`;
